source/usage-examples/updateOne.txt

```diff
@@ -0,0 +1,118 @@
+of ``UpdateResult``.  You can retrieve information such as the number of
+documents modified calling the ``getModifiedCount()`` method, or the
+value of ``_id`` field by calling the ``getUpsertedId()`` method if you
+specified ``upsert(true)`` in the ``UpdateOptions``.
+
+If your update operation fails, the driver raises an exception.
+For example, if you try to set a value for the immutable field ``_id`` in
+your update document, the method throws a ``MongoWriteException`` with the
+message:
+
+.. code-block:: none
+
+   Performing an update on the path '_id' would modify the immutable field '_id'
+
+If your update document contains a change that violates unique index
+rules, the method throws a ``DuplicateKeyException`` with the error
+message that resembles the following:
+
+.. code-block:: none
+
+   E11000 duplicate key error collection: ...
+
+For more information on the types of exceptions raised under specific
+conditions, see the API documentation for ``updateOne()``, linked at the
+bottom of this page.
+
+For more information on updating documents using the MongoDB Java driver,
+see our guide (TODO: Fundamentals > CRUD Operations > Write Operations)
+on updating documents.
+
+Example
+-------
+
+In this example, we update the first match for our query in the ``movies``
+collection of the ``sample_mflix`` database. We perform the following
+updates to the matching document:
+
+- Set the value of ``runtime`` to ``99``
+- Add ``Sports`` to the array of ``genres`` only if it does not already
+  exist
+- Set the value of ``lastUpdated`` to the current time.
+
+We use the ``Updates`` builder, a factory class that contains static
+helper methods to construct the update document. While you can pass an update
+document instead of using the builder, the builder provides type checking and
+simplified syntax. Read our Updates guide (TODO link) in the Builders section
+for more information.
+
+.. include:: /includes/connect-guide-note.rst
+
+.. literalinclude:: /includes/usage-examples/code-snippets/UpdateOne.java
+   :language: java
+
+If you are using the **legacy API**, see (TODO: FAQ ?) to determine what
+changes you need to make to this code example.
+
+After you run the example, you should see output that resembles the
+following:
+
+.. code-block:: none
+
+   Modified document count: 1
+
+If you query the updated document, it should resemble the following:
+
+.. code-block:: none
+
+   Document {
+     { _id=...,
+       plot=...,
+       genres=[Adventure, Comedy, Family, Sports],
+       runtime=99,
+       ...
+       lastUpdated=Timestamp{...}
+     }
+   }
+
+For additional information on the classes and methods mentioned on this
+page, see the following API documentation:
+
+- :java-sync-api:`UpdateOne <com/mongodb/client/MongoCollection.html#updateOne(com.mongodb.client.ClientSession,org.bson.conversions.Bson,java.util.List)>`
+- :java-sync-api:`UpdateOptions <com/mongodb/client/model/UpdateOptions.html>`
```

Comment:
As I commented above, all of these links should also be inline links above, where they first appear.